src/Database/SqlDialectTrait.php

```diff
@@ -206,6 +206,14 @@ protected function _updateQueryTranslator($query)
      */
     protected function _removeAliasesFromConditions($query)
     {
+        if (!empty($query->clause('join'))) {
+            trigger_error(
```

Comment:
@jippi I've settled for a notice as I thought introducing exceptions in patch releases would be discouraged?, and because the only purpose of this notice is to inform developers that are trying to implement join support.